bin/gridsub.py

```diff
@@ -66,22 +74,99 @@ class JSON(str):
-def cli(campaign, details, file, jobfile, json, jsonjob):
+def cli(campaign, details, file, jobfile, jsonCLI, jsonjob):
     """
     This command allow the user to submit campaigns to CiGri.
     """
 
+    client = ApiClient(
+        host="localhost",
+        port=4430,
+        baseEndpoint="/cigri-api",
+        headers={
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        },
+    )
+
+    gridsub(campaign, details, file, jobfile, jsonCLI, jsonjob, client)
+
+
+def submit_campaign(jdl, client, campaign_id=None):
+    url = "/campaigns"
+
+    if campaign_id is not None:
+        url += f"/{campaign_id}/jobs"
+
+    return client.post(url, body=jdl)
+
+
+def gridsub(campaign, details, file, jobfile, jsonCLI, jsonjob, client):
     # Checking option content
 
-    if file is None and jobfile is None and json is None and jsonjob is None:
+    if (
+        file is None
+        and jobfile is None
+        and jsonCLI is None
+        and jsonjob is None
+    ):
         raise click.MissingParameter(
             "Must provide one of the four --file, --jobfile, --json, --jsonjob option",
             param_type="option",
         )
 
+    if (jobfile is not None or jsonjob is not None) and campaign is None:
+        raise click.MissingParameter(
+            "Must provide a campaign id using --campaign when using --jobfile or --jsonjob",
+            param_type="option",
+        )
+
     # Function implementation
 
-    raise RuntimeError("NYI")
+    jsons = []
+    jobs = []
+
+    if file is not None:
+        with open(file) as f:
+            jsons.append(f.read())
+
+    if jobfile is not None:
+        with open(jobfile) as f:
+            jobs.append(f.read())
+
+    if jsonCLI is not None:
+        jsons.append(jsonCLI)
+
+    if jsonjob is not None:
+        jobs.append(jsonjob)
+
+    for json_ in jsons:
+        response = submit_campaign(json_, client)
+        body = response.read()
+        if response.status == 201:
+            print("Campaign successfully submitted")
+            if details:
+                print(json.dumps(json.loads(body.decode()), indent=2))
+        else:
+            try:
+                error = json.loads(body)
+                if error["title"] == "Admission rule error":
+                    print(f"Admission rule error:\n\t{error['message']}")
+                else:
+                    raise RuntimeError(f"Error: {response.body}")
+            except json.decoder.JSONDecodeErrors:
+                raise RuntimeError(f"Error: {body}")
+
+    for job in jobs:
+        response = submit_campaign(job, client, campaign_id=campaign)
+        body = response.read()
+
+        if response.status == 201:
+            print(f"Jobs successfully added to campaign {campaign}")
+            if details:
+                print(json.dumps(json.loads(body.decode()), indent=2))
+        else:
+            raise RuntimeError(f"Error submitting campaign: {body}")
```

Comment:
submitting campaign ? Shouldn't it be submitting job ?

Reply:
I fixed the error message in the last commit